src/aks-preview/azext_aks_preview/tests/latest/test_decorator.py

```diff
@@ -887,13 +887,103 @@ def test_get_disk_driver_update(self):
         )
         mc = self.models.ManagedCluster(
             location="test_location",
-            storage_profile=storage_profile,
         )
         ctx_5.attach_mc(mc)
         self.assertEqual(
             ctx_5.get_storage_profile(), storage_profile
         )
 
+        # disk_driver_version value passed and enable-disk-driver passed
+        ctx_6 = AKSPreviewContext(
+            self.cmd,
+            {
+                "enable_disk_driver": True,
+                "disk_driver_version": "v2",
+                "enable_file_driver": True,
+                "enable_snapshot_controller": True,
+            },
+            self.models,
+            decorator_mode=DecoratorMode.UPDATE,
+        )
+        storage_profile = (
+            self.models.ManagedClusterStorageProfile(
+                disk_csi_driver = self.models.ManagedClusterStorageProfileDiskCSIDriver(
+                    enabled = True,
+                    version = "v2",
+                ),
+                file_csi_driver = self.models.ManagedClusterStorageProfileFileCSIDriver(
+                    enabled = True,
+                ),
+                snapshot_controller = self.models.ManagedClusterStorageProfileSnapshotController(
+                    enabled = True,
+                ),
+            )
+        )
+        mc = self.models.ManagedCluster(
+            location="test_location",
+        )
+        ctx_6.attach_mc(mc)
+        self.assertEqual(
+            ctx_6.get_storage_profile(), storage_profile
+        )
+
+        # fail with enable-disk-driver as false and value passed for disk_driver_version
+        ctx_8 = AKSPreviewContext(
+            self.cmd,
+            {
+                "disable_disk_driver": True,
+                "disk_driver_version": "v2",
+            },
+            self.models,
+            decorator_mode=DecoratorMode.UPDATE,
+        )
+
+        # fail on argument usage error
+        with self.assertRaises(ArgumentUsageError):
+            ctx_8.get_disk_driver()
+
+        #  enable-disk-driver, disk_driver_version, enable_file_driver and enable_snapshot_controller passed
+        ctx_9 = AKSPreviewContext(
+            self.cmd,
+            {
+                "enable_disk_driver": True,
+                "disk_driver_version": "v2",
+                "disable_file_driver": True,
+                "disable_snapshot_controller": True,
+            },
+            self.models,
+            decorator_mode=DecoratorMode.UPDATE,
+        )
+        storage_profile = (
+            self.models.ManagedClusterStorageProfile(
+                disk_csi_driver = self.models.ManagedClusterStorageProfileDiskCSIDriver(
+                    enabled = True,
+                    version = "v2",
+                ),
```

Comment:
From existing code, this should be something like,
```suggestion
                disk_csi_driver = self.models.ManagedClusterStorageProfileDiskCSIDriver(
                    enabled = True,
                    version = "v2",
                ),
                file_csi_driver = self.models.ManagedClusterStorageProfileFileCSIDriver(
                    enabled = False,
                ),
                snapshot_controller = self.models.ManagedClusterStorageProfileSnapshotController(
                    enabled = False,
                ),
```

Reply:
Fixed